pkg/runtime/scheduler/scheduler.go

```diff
@@ -136,147 +154,218 @@ func (m *Manager) processStream(ctx context.Context, client *client.Client, stre
-						}
-
-						if nerr := stream.CloseSend(); nerr != nil {
-							log.Errorf("Error closing stream for Scheduler address: %s", client.Address)
-						}
-						if nerr := client.CloseConnection(); nerr != nil {
-							log.Errorf("Error closing conn for Scheduler address %s: %v", client.Address, nerr)
-						}
-						return err
+			}
+		}
+	}()
+
+	for {
+		select {
+		case <-stream.Context().Done():
+			if err := stream.CloseSend(); err != nil {
+				log.Errorf("Error closing stream for Scheduler address %s: %v", client.Address, err)
+			}
+			if err := client.CloseConnection(); err != nil {
+				log.Errorf("Error closing conn for Scheduler address %s: %v", client.Address, err)
+			}
+			return stream.Context().Err()
+		case <-ctx.Done():
+			if err := stream.CloseSend(); err != nil {
+				log.Errorf("Error closing stream for Scheduler address %s", client.Address)
+			}
+			if err := client.CloseConnection(); err != nil {
+				log.Errorf("Error closing connection for Scheduler address %s: %v", client.Address, err)
+			}
+			return ctx.Err()
+		default:
+			resp, err := stream.Recv()
+			if err == nil {
+				m.wg.Add(1)
+				go func() {
+					defer m.wg.Done()
+					select {
+					case <-ctx.Done():
+					case m.jobCh <- resp:
 					}
+				}()
+
+				continue
+			}
+
+			switch status.Code(err) {
+			case codes.Canceled:
+				log.Errorf("Sidecar cancelled the stream ctx for Scheduler address %s.", client.Address)
+			case codes.Unavailable:
+				log.Errorf("Scheduler cancelled the stream ctx for address %s.", client.Address)
+			default:
+				if err == io.EOF {
+					log.Errorf("Scheduler cancelled the Sidecar stream ctx for Scheduler address %s.", client.Address)
+				} else {
+					log.Errorf("Error while receiving job trigger from Scheduler at address %s: %v", client.Address, err)
 				}
 
-				log.Debugf("Received job: %+v %+v", resp.GetData(), resp.GetMetadata())
-				err = m.invokeAppMethod(ctx, resp)
-				if err != nil {
-					log.Errorf("Error invoking app method: %v", err)
+				if nerr := stream.CloseSend(); nerr != nil {
+					log.Errorf("Error closing stream for Scheduler address: %s", client.Address)
 				}
+				if nerr := client.CloseConnection(); nerr != nil {
+					log.Errorf("Error closing conn for Scheduler address %s: %v", client.Address, nerr)
+				}
+				return err
 			}
 		}
 	}
 }
 
-func (m *Manager) invokeAppMethod(ctx context.Context, resp *schedulerv1pb.WatchJobsResponse) error {
-	parts := strings.Split(resp.GetName(), "||")
-	jobName := parts[len(parts)-1]
-
-	switch m.isHTTP {
-	case true:
-		req := v1.NewInvokeMethodRequest("dapr/receiveJobs/"+jobName).
-			WithHTTPExtension(http.MethodPost, "").
-			WithDataObject(resp.GetData())
-		if req != nil {
-			defer req.Close()
+func (m *Manager) processTriggers(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case job := <-m.jobCh:
+			log.Debugf("Received scheduled Job")
```

Comment:
Do you think its helpful to add the job name back here? It might be helpful for debugging if there are issues to know what job was received vs just received one of probably many jobs and have to guess which one
```suggestion
			log.Debugf("Received scheduled job")
```